org.eclipse.scout.rt.rest.jersey.client/src/main/java/org/eclipse/scout/rt/rest/jersey/client/ChunkedDataReader.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2010, 2025 BSI Business Systems Integration AG
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.scout.rt.rest.jersey.client;
+
+import static org.eclipse.scout.rt.platform.util.Assertions.*;
+
+import java.io.IOException;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+
+import jakarta.ws.rs.core.GenericType;
+import jakarta.ws.rs.core.Response;
+
+import org.eclipse.scout.rt.platform.util.StringUtility;
+import org.eclipse.scout.rt.rest.client.chunked.IChunkedDataReader;
+import org.glassfish.jersey.client.ChunkedInput;
+
+/**
+ * Scout REST API wrapper around Jersey {@link ChunkedInput} instance.
+ * <p>
+ * See documentation
+ * <a href="https://eclipse-ee4j.github.io/jersey.github.io/documentation/latest/user-guide.html#d0e10936">Jersey chunked input</a>
+ */
+public class ChunkedDataReader<T> implements IChunkedDataReader<T> {
+
+  protected ChunkedInput<T> m_input;
+
+  @Override
+  public void init(Response response, Class<T> type, String delimiter) {
+    assertNull(m_input, "already initialized");
+    assertNotNull(delimiter, "missing delimiter");
```

Comment:
Are you sure that this assertion is correct? In IChunkedDataReader.create(Response, Class<T>), the delimiter is set to null, which would then trigger the assertion. Also if null was forbidden, an empty string would have to be used to activate the default delimiter, which does not feel right. And the use of StringUtility.isNullOrEmpty() would be unnecessary.